core/src/main/java/org/elasticsearch/index/query/OrQueryBuilder.java

```diff
@@ -74,7 +98,66 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        if (filters.isEmpty()) {
+            // no filters provided, this should be ignored upstream
+            return null;
+        }
+
+        BooleanQuery query = new BooleanQuery();
+        for (QueryBuilder f : filters) {
+            query.add(f.toQuery(parseContext), Occur.SHOULD);
```

Comment:
Good catch, just copied that part of the code but know we have this extra step. I think if can return `null`. This is anoying, maybe at some point we can revisit this and see if instead of passing up `null` up through the whole query-tree we can have some sort of No-Op Query Builder and maybe an equivalent as a lucene query. Just a thought, will just catch this here.